sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CoalesceShufflePartitions.scala

```diff
@@ -79,8 +66,29 @@ case class CoalesceShufflePartitions(session: SparkSession) extends Rule[SparkPl
           case stage: ShuffleQueryStageExec if stageIds.contains(stage.id) =>
             CustomShuffleReaderExec(stage, partitionSpecs)
         }
+      }
+
+      if (validMetrics.isEmpty) {
```

Comment:
if a query stage has multiple leaf shuffles, and only one of them has 0-partition input RDD. What shall we do?

Reply:
I think it's like coalescing one less shuffles and handled by the `nonEmpty` codes.